BinaryObjectScanner/Protection/SmartE.cs

```diff
@@ -1,28 +1,84 @@
+           if (pex.InternalName.OptionalEquals("SmarteSECURE"))
+           {
+               return "SmartE";
+           }
+           var sections = pex.Model.SectionTable ?? [];
 
-            return null;
-        }
+           if (sections.Length > 0)
+           {
+               // Get the last section data, if it exists
+               var lastSectionData = pex.GetSectionData(sections.Length - 1);
+               if (lastSectionData != null)
+               {
+                   // All sections seen so far are the last sections, so this is "technically"
+                   // the only known needed check so far. Others kept as backups if this fails
+                   // on some future entry
+                   var matchers = GenerateMatchers();
+                   var match = MatchUtil.GetFirstMatch(file, lastSectionData, matchers, includeDebug);
+                   if (!string.IsNullOrEmpty(match))
+                       return match;
+               }
+           }
+           
+           // Get the .bss section, if it exists
+           var bssSectionRaw = pex.GetFirstSectionData(".bss", exact: true);
+           if (bssSectionRaw != null)
+           {
+               //Rise of Nations
+               var matchers = GenerateMatchers();
+               var match = MatchUtil.GetFirstMatch(file, bssSectionRaw, matchers, includeDebug);
+               if (!string.IsNullOrEmpty(match))
+                   return match;
+           }
 
+           // Get the .tls section, if it exists
+           var tlsSectionRaw = pex.GetFirstSectionData(".tls", exact: true);
+           if (tlsSectionRaw != null)
+           {
+               //Zoo Tycoon 2
+               var matchers = GenerateMatchers();
+               var match = MatchUtil.GetFirstMatch(file, tlsSectionRaw, matchers, includeDebug);
+               if (!string.IsNullOrEmpty(match))
+                   return match;
+           }
+           
+           // Get the .idata section, if it exists
+           var idataSectionRaw = pex.GetFirstSectionData(".idata", exact: true);
+           if (idataSectionRaw != null)
+           {
+               // http://redump.org/disc/58561/ and http://redump.org/disc/71983/
+               var matchers = GenerateMatchers();
+               var match = MatchUtil.GetFirstMatch(file, idataSectionRaw, matchers, includeDebug);
+               if (!string.IsNullOrEmpty(match))
+                   return match;
+           }
+
+           // Get the .edata section, if it exists
+           var edataSectionRaw = pex.GetFirstSectionData(".edata", exact: true);
+           if (edataSectionRaw != null)
+           {
+               // http://redump.org/disc/36619/
+               var matchers = GenerateMatchers();
+               var match = MatchUtil.GetFirstMatch(file, edataSectionRaw, matchers, includeDebug);
+               if (!string.IsNullOrEmpty(match))
+                   return match;
+           }
```

Comment:
Please add an extra newline after this curly brace.

Reply:
Irrelevant due to https://github.com/SabreTools/BinaryObjectScanner/pull/353#discussion_r1914165467

Reply:
Fixed.